### PR TITLE
Making moving furniture into an activity actor

### LINKED ISSUE #45010: Not Interrupted while Grabbing and moving furniture
# Describe the bug

When you `G`rab some furniture and try to move it, the character is not interrupted if approached or attacked by an enemy

# Steps To Reproduce

1. `G`rab something heavy (fridge, counter, bookcase...)
2. Drag it towards a group of enemies
3. Don't get asked to stop when they approach/attack you

# Expected behavior

The player gets asked if they wish to stop the activity like any other activity when an enemy is nearby/attacking them.

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.E-7041-gdb7d137 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Generic Guns [generic_guns],
    Generic Guns - UndeadPeople Tileset [Generic_GunsUDP],
    Graphical Overmap [Graphical_Overmap],
    Bionic Slots [cbm_slots],
    Stats Through Skills [StatsThroughSkills]
]

# Additional context

I assume the Grabbing mechanics need to be moved to a `player_activity` so they can be properly interruptable.

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
SUMMARY: Bugfixes "Making moving furniture into an activity actor"

#### Purpose of change
Fixes #45010
Allow interruptions while moving furniture.

#### Describe the solution
Added new activity actor, broke up function grabbed_furn_move into one that finds the moves and other that does the rest and had activity actor call them.

### CODE REVIEW

File: src/activity_actor.cpp [hunk 1]
```diff
@@ -2208,6 +2208,47 @@ std::unique_ptr<activity_actor> stash_activity_actor::deserialize( JsonIn &jsin
...
+    if( !g->grabbed_furn_move( dp ) ) {
+        g->walk_move( who.pos() + dp, via_ramp, true );
+    }
+    act.set_to_null();
+}
+
+void move_furniture_activity_actor::canceled( player_activity &, Character &who )
+{
+    add_msg( m_warning, _( "You let go of the grabbed object." ) );
+    get_avatar().grab( object_type::NONE );
```
Comment (anothersimulacrum) [inline src/activity_actor.cpp:2229]:
```suggestion
    who.grab( object_type::NONE );
```

File: src/activity_actor.cpp [hunk 2]
```diff
@@ -2208,6 +2208,47 @@ std::unique_ptr<activity_actor> stash_activity_actor::deserialize( JsonIn &jsin
...
+{
+    jsout.start_object();
+    jsout.member( "dp", dp );
+    jsout.member( "via_ramp", via_ramp );
+    jsout.end_object();
+}
+
+std::unique_ptr<activity_actor> move_furniture_activity_actor::deserialize( JsonIn &jsin )
+{
+    move_furniture_activity_actor actor = move_furniture_activity_actor( tripoint( 0, 0, 0 ), false );
```
Comment (anothersimulacrum) [inline src/activity_actor.cpp:2242]:
```suggestion
    move_furniture_activity_actor actor = move_furniture_activity_actor( tripoint_zero, false );
```

File: src/game.cpp [hunk 3]
```diff
@@ -10223,6 +10224,91 @@ bool game::phasing_move( const tripoint &dest_loc, const bool via_ramp )
...
+    const bool canmove = (
+                             m.passable( fdest ) &&
+                             critter_at<npc>( fdest ) == nullptr &&
+                             critter_at<monster>( fdest ) == nullptr &&
+                             ( !pulling_furniture || is_empty( u.pos() + dp ) ) &&
+                             ( !has_floor || m.has_flag( "FLAT", fdest ) ) &&
+                             !m.has_furn( fdest ) &&
+                             !m.veh_at( fdest ) &&
+                             ( !has_floor || m.tr_at( fdest ).is_null() )
+                         );
```
Comment (anothersimulacrum) [inline src/game.cpp:10254]: This might look nicer without the outer parentheses.

File: src/activity_actor.cpp (hunk 1 above)
Comment (Ramza13) [reply src/activity_actor.cpp:2229]: grab doesn't exist in Character.  I don't know that it makes sense to move it there.

Comment (anothersimulacrum) [reply src/activity_actor.cpp:2229]: Then you should ditch the `who` argument.

**Pull request merged**